test/controllers/articles_controller_test.rb

```diff
@@ -91,86 +132,53 @@ def test_edit_article_post
       article: { title: new_title, body: new_body }
     }
     login(users(:zero_user).login)
-    post(:edit_article, params)
+    post(:update, params)
 
     assert_flash_text(:permission_denied.l)
-    assert_redirected_to(action: :index_article)
+    assert_redirected_to(articles_path)
 
     # Prove authorized user can edit article
     login(users(:article_writer).login)
     make_admin
-    post(:edit_article, params)
+    post(:update, params)
     article.reload
 
     assert_flash_success
-    assert_redirected_to(action: :show_article, id: article.id)
+    assert_redirected_to(article_path(article.id))
     assert_equal(new_title, article.title)
     assert_equal(new_body, article.body)
 
     # Prove that saving without changes provokes warning
     # save it again without changes
-    post(:edit_article, params)
+    post(:update, params)
     article.reload
     assert_flash_warning
-    assert_redirected_to(action: :show_article, id: article.id)
+    assert_redirected_to(article_path(article.id))
 
     # Prove removing title provokes warning
     params[:article][:title] = ""
-    post(:edit_article, params)
+    post(:update, params)
     assert_flash_text(:article_title_required.l)
-    assert_template(:edit_article)
+    assert_template(:edit)
```

Comment:
Replace with `assert_select` that tests for presence of form that posts to `:edit`.